File: src/Java/Algs4LibraryPractice/AccumulatorExample (Inheritance)/Accumulator.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Iterator;

public class Accumulator {
	private int count;
	private double sum;

	// Default constructor will initialize both data-type values.

	public void addDataValue(double value) {
		sum += value;
		count++;
	}

	public double mean() {
		return sum / count;
	}

	public String toString() {
		return "Mean (" + count + " values): " + String.format("%7.5f", mean());

	}

	public int getCount() {
		return count;
	}

	public double getSum() {
		return sum;
	}
	
	// Test Client
	public static void main(String[] args) {
		BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
		Iterator<String> inputIterator = input.lines().iterator();
		Accumulator accumulator = new Accumulator();

		while (inputIterator.hasNext()) {
			accumulator.addDataValue(Double.parseDouble(inputIterator.next()));
		}
		System.out.println(accumulator);
	}
}

File: src/Kotlin/Algs4LibarayPractice/Accumulator.kt
package Kotlin.Algs4LibarayPractice

import edu.princeton.cs.algs4.StdRandom

class Accumulator(private var sum: Double, private var count: Int) {

    fun addDataValue(value: Double) {
        sum += value;
        count++;
    }

    fun mean(): Double {
        return sum / count;
    }

    override fun toString(): String {
        return "Mean $count values: " + String.format("%7.5f", mean());
    }
}

// Test Client
fun main(args: Array<String>) {
    //var trials = args[0].toInt();
    val trials = 10;
    var accumulator = Accumulator(0.0, 0);

    for (i: Int in 0..trials) {
        accumulator.addDataValue(StdRandom.uniform());
    }
    println(accumulator);
}